packages/@aws-cdk/custom-resource-handlers/lib/custom-resources-framework/classes.ts

```diff
@@ -353,4 +363,38 @@ export abstract class HandlerFrameworkClass extends ClassType {
     const superInitializerArgs: Expression[] = [scope, id, props.superProps];
     init.addBody(new SuperInitializer(...superInitializerArgs));
   }
+
+  private buildCustomResourceRuntimeDeterminer() {
+    const runtimeDeterminer = this.addMethod({
+      name: 'builtInCustomResourceNodeRuntime',
+      visibility: MemberVisibility.Private,
+      static: true,
+      returnType: LAMBDA_MODULE.Runtime,
+    });
+    runtimeDeterminer.addParameter({
+      name: 'scope',
+      type: CONSTRUCTS_MODULE.Construct,
+    });
+    runtimeDeterminer.addBody(
+      stmt.constVar(expr.ident('runtimeName'), expr.directCode('core.Stack.of(scope).regionalFact(FactName.DEFAULT_CR_NODE_VERSION, "nodejs18.x")')),
```

Comment:
There is an API somewhere that will do a synth-time lookup if the region is known at synth time, or generate a runtime table for an agnostic stack. Not sure where it is again, but look it up.

Reply:
Yeah `regionalFact` is the right API.

Put the hardcoded `18.x` fallback in an obvious place somewhere, right next to the other table probably.